### ISSUE TITLE
Proposal: a few breaking changes

### ISSUE THREAD

**Issue by ziotom78** [litebird/litebird_sim#65]
I am proposing two breaking changes in the code that might improve the consistency of some parts.

# Rename `Instrument` to `InstrumentInfo`

There is a class named `DetectorInfo` in the framework, which closely matches the quantity `detector_info` in the IMO, and PR #60 is going to include a class `FreqChannelInfo`, which matches the quantity `channel_info`. As the `Instrument` class (currently used by the scanning strategy simulator) is meant to match the newly created `instrument_info` IMO quantity, for consistency it would be advisable to rename `Instrument` to `InstrumentInfo`.

Of course, this introduces a breaking change in the code, which will require to bump the version number from 0.1.0 to 0.2.0.

# Angles are always expressed in radians

The current implementation of the `SpinningScanningStrategy` class accepts angles *in degrees* in the `__init__` constructor; these are immediately converted in radians for efficiency reasons (NumPy and Numba trigonometric functions work with radians, so it's better to measure every angle using this unit). However, this leads to a few weirdnesses:

```python
import litebird_sim as lbs
import numpy as np

sstr = lbs.SpinningScanningStrategy(
    # Specify the angle in degrees and spare an explicit call to np.deg2rad
    spin_sun_angle_deg=10.0,
    ...
)

# Surprise! There is no "spin_sun_angle_deg" field in "sstr"!
print(sstr.spin_sun_angle_rad) # You get the angle in radians

# If I want to modify the angle, I must now use "deg2rad" explicitly
sstr.spin_sun_angle_rad += np.deg2rad(5.0)
```

When I implemented `SpinningScanningStrategy`, I chose to use degrees in the constructor because angles in the IMO are expressed in this unit for the sake of readability and to avoid roundoff errors (when expressed in degrees, most of them are integer numbers).

However, while implementing PR #60, I had a similar problem for the `InstrumentInfo` class (which contains the angles specifying the orientation of the focal plane). This time I chose an alternate and more straightforward approach, and I required to use *radians* even in the constructor. (The only place where degrees are used is just when reading an object from the IMO, because of course angles in the IMO are in degrees!) Therefore, here is the code to create a `InstrumentInfo` object:

```python
import numpy as np

instr = lbs.InstrumentInfo(
    spin_boresight_angle_rad=np.deg2rad(10.0),
    ...
)

# No surprise here: you get what you gave
print(instr.spin_boresight_angle_rad)

# We have to use deg2rad, but it's ok, as it is consistent with the code above
instr.spin_boresight_angle_rad += np.deg2rad(5.0)
```

The more I use this latter implementation, the more I find it pleasant. It's true that you always must call `np.deg2rad`, but it has a few advantages:

1.  The keywords used for `__init__` are the same as the fields of the class, and this prevent surprises;
2.  If you need to modify the angle later, you keep using the same units as in the constructor;
3.  It's not a real PITA, as most of the time people create these objects using `InstrumentInfo.from_imo`, which performs the degree-to-radians conversion automatically (the IMO reports these angles in degrees).

Being a breaking change, this will require a version bump too, but we might take two birds with one stone and implement the two changes together for 0.2.0.

Basically, the proposal is to change the declaration of `SpinningScanningStrategy.__init__` so that it always accepts only angles in radians. (Thus, keyword names will be changed from `spin_sun_angle_deg` to `spin_sun_angle_rad`, etc.)

I would like to hear your opinions!

**Comment by mreineck** [comment]
I'm very much in favor of always using angles in radians, at least within the code! (Simply because that's the common unit when calling trigonometric functions.)

LevelS used this convention as well, and I think it really helped avoid confusion.
(It still took angles in degrees or arc minutes as parameters in specific circumstances where there was a strong convention, but these were immediately converted after parsing.)

**Comment by paganol** [comment]
I like that, it is much cleaner. It is true that most of the time users will populate `InstrumentInfo` or `SpinningScanningStrategy ` with `from_imo` and all of this will be totally transparent for them, but still better have everything in radians.

**Comment by dpole** [comment]
I second both the proposals!

**Comment by ziotom78** [comment]
This is done by PR #60 and #69.
